tests/pytests/unit/modules/test_chroot.py

```diff
@@ -0,0 +1,328 @@
+"""
+:maintainer:    Alberto Planas <aplanas@suse.com>
+:platform:      Linux
+"""
+
+import sys
+
+import pytest
+
+import salt.loader.context
+import salt.modules.chroot as chroot
+import salt.utils.platform
+from salt.exceptions import CommandExecutionError
+from tests.support.mock import MagicMock, patch
+
+pytestmark = [
+    pytest.mark.skipif(
+        salt.utils.platform.is_windows(), reason="This test cannot work on Windows"
+    )
+]
+
+
+@pytest.fixture
+def configure_loader_modules():
+    loader_context = salt.loader.context.LoaderContext()
+    return {
+        chroot: {
+            "__salt__": {},
+            "__utils__": {"files.rm_rf": MagicMock()},
+            "__opts__": {"extension_modules": "", "cachedir": "/tmp/"},
+            "__pillar__": salt.loader.context.NamedLoaderContext(
+                "__pillar__", loader_context, {}
+            ),
+        }
+    }
+
+
+def test__create_and_execute_salt_state():
+    with patch("salt.client.ssh.wrapper.state._cleanup_slsmod_low_data", MagicMock()):
+        with patch("salt.utils.hashutils.get_hash", MagicMock(return_value="deadbeaf")):
+            with patch("salt.fileclient.get_file_client", MagicMock()):
+                with patch("salt.modules.chroot.call", MagicMock()):
+                    chroot._create_and_execute_salt_state("", {}, {}, False, "md5")
+
+
+@patch("os.path.isdir")
+def test_exist(isdir):
+    """
+    Test if the chroot environment exist.
+    """
+    isdir.side_effect = (True, True, True, True)
+    assert chroot.exist("/chroot")
+
+    isdir.side_effect = (True, True, True, False)
+    assert not chroot.exist("/chroot")
```

Comment:
```suggestion
def test_exist():
    """
    Test if the chroot environment exist.
    """
    with patch("os.path.isdir") as isidr:
        isdir.side_effect = (True, True, True, True)
        assert chroot.exist("/chroot")

    with patch("os.path.isdir") as isidr:
        isdir.side_effect = (True, True, True, False)
        assert not chroot.exist("/chroot")
```
And the same for all `@patch` decorators please

Reply:
Thanks Pedro